src/OICodeGen.cpp

```diff
@@ -986,6 +989,99 @@ std::string OICodeGen::typeToName(drgn_type *type) {
   return typeName;
 }
 
+bool OICodeGen::recordChildren(drgn_type *type) {
+  drgn_type_template_parameter *parents = drgn_type_parents(type);
+
+  for (size_t i = 0; i < drgn_type_num_parents(type); i++) {
+    struct drgn_qualified_type t {};
+
+    if (auto *err = drgn_template_parameter_type(&parents[i], &t)) {
+      LOG(ERROR) << "Error when looking up parent class for type " << type
+                 << " err " << err->code << " " << err->message;
+      drgn_error_destroy(err);
+      continue;
+    }
+
+    drgn_type *parent = drgnUnderlyingType(t.type);
+    if (!isDrgnSizeComplete(parent)) {
+      continue;
+    }
+
+    const char *parentName = drgn_type_tag(parent);
+    if (!parentName) {
+      continue;
+    }
+
+    // drgn pointers are not stable, so use string representation for reverse
+    // mapping for now. We need to find a better way of creating this
+    // childClasses map - ideally drgn would do this for us.
+    childClasses[parentName].push_back(type);
+    VLOG(1) << drgn_type_tag(type) << "(" << type << ") is a child of "
+            << drgn_type_tag(parent) << "(" << parent << ")";
+  }
+
+  return true;
+}
+
+/*
+ * Build a mapping of Class -> Children
+ *
+ * drgn only gives us the mapping Class -> Parents, so we must iterate over all
+ * types in the program to build the reverse mapping.
+ */
+bool OICodeGen::enumerateChildClasses() {
+  if (!config.polymorphicInheritance) {
+    return true;
+  }
+
+  if ((setenv("DRGN_ENABLE_TYPE_ITERATOR", "1", 1)) < 0) {
```

Comment:
Should we remove this? We have to set this env variable mid program in a few places and it doesn't come from upstream. `setenv` called like this can technically only fail with `ENOMEM` so it should be fine either way though.

Reply:
I don't see the point in this env variable either, but I'll leave it as-is for now to avoid bloating this PR. We might want to change drgn to remove it in the future.